packages/@ember/-internals/utils/types.ts

```diff
@@ -1,7 +1,15 @@
 export type AnyFn = (...args: any[]) => any;
 
+export type MethodsOf<O> = {
+  [K in keyof O]: O[K] extends AnyFn ? O[K] : never;
+};
```

Comment:
This doesn't actually work, unfortunately. You need something like the final `keyof` as used in `MethodNamesOf` below to make it properly strip out the bits which are actually `never`. It turns out the easiest way to get `MethodsOf` is to reuse `MethodNamesOf` in conjunction with `Pick`:

```ts
type MethodsOf<O> = Pick<O, MethodNamesOf<O>>;
```

[playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggdiAYnKBeKAKAdDghgJwHMBnALilwQG0BdASjQD4KEBuAKHYgA8wB7fMCihIUALIRgACz4ATYgHkAZgB4FzdAG92UKFQDSUAJYoA1hBB8lUBTXIKDNKD2AQ482AmRQA-DcdQ5HAQAG4Q+BwAvhzsItAAQpKu+BLScoqq6mhQAApGAMamagA04pIysgByuAC2EBlqjIwxPPyCwuDQqRXVdQ0AKhpQ2rp6hiZQ5pbW-XZQ-QEubh7wSCh+hkGh4RxQkVRTVvM0MQD0p1AAVgCuxELEfFCW11D5lFDEENAyAO7GQj8BKZiFAfkZpMIpNAeLUwAAbepQABGEDhfB+7FkEHycII0BxuGIILEuBAKJUg2GOlGIVwcOuEB85H6u1GNVwYBUAFVGBglHByBhgMyGKhmFy6OQSWSINzmuxIpw4lAABrZbpyXr1ZQqaXk65wUxwdFwJocZUATXV5XSOr1soNRpNZtinSgAC1solgMkNfI7aT9YbjT9Tc0gA)